src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,227 @@
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onNavigate: PropTypes.func,
+
+    /** The report currently being looked at */
+    report: reportPropTypes,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+};
+
+const defaultProps = {
+    sourceURL: '',
+    report: {
+        isLoadingMoreReportActions: false,
+        reportID: '',
+    },
+    reportActions: {},
+    onNavigate: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.makeStateWithReports = this.makeStateWithReports.bind(this);
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+        this.onShowArrow = this.onShowArrow.bind(this);
+
+        this.state = {shouldShowArrow: this.canUseTouchScreen};
+    }
+
+    componentDidMount() {
+        this.makeStateWithReports();
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeStateWithReports();
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    onShowArrow(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const sourceURL = _.get(attachmentItem, 'sourceURL', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        this.props.onNavigate({sourceURL: addEncryptedAuthTokenToURL(sourceURL), file});
+
+        return {
+            sourceURL,
+            file,
+        };
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actionsArr = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+
+        /**
+         * calling reducer will filter out attachments, determine the index of opened attachment,
+         * and retrieve the src url and name of attachements
+         */
+        const attachments = _.reduce(actionsArr, (attachmentsAccumulator, {originalMessage}) => {
+            if (originalMessage && originalMessage.html) {
+                const matchesIt = originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA);
+                const matches = [...matchesIt];
```

Comment:
Can we make this one line?

```suggestion
                const matches = [...originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA];
```